doc/Developer-Guide.md

```diff
@@ -0,0 +1,453 @@
+
+Click `Create & Open` button and when workspace starts up you'll observe a workspace with IntelliJ Idea Community Edition:
+
+![intellij-idea-community-edition](https://raw.githubusercontent.com/che-incubator/che-editor-intellij-community/master/doc/images/intellij-idea-community-edition.jpg)
+
+##### Build IntelliJ Idea Ultimate Edition
+
+To build Intellij Idea Ultimate Edition additional parameter should be passed through build argument:
+
+```
+PRODUCT_NAME=ideaIU
+```
+
+> Note, that product name should passed in the same way how it is provided in current example. To see mapping JetBrains product to its product name see section: [JetBrains product name mapping](#jetbrains-product-name-mapping)
+
+Build command will look like this:
+
+```
+$ docker build -t idea-iu --build-arg PRODUCT_NAME=ideaIU .
+```
+
+This command will build image with **2020.2.3** version by default. Built image can be tagged and pushed to user repository the same way how it is described in previous section:
+
+```
+$ docker tag idea-iu:latest <username>/idea-iu:latest
+$ docker push <username>/idea-iu:latest
+```
+
+The next step is to provision activation code for offline usage. To be able to use IntelliJ Idea Ultimate Edition with registered license. See section: [Provision activation code for offline usage](#provision-activation-code-for-offline-usage)
+
+Now it is time to create workspace with the following `workspace.yaml` and `meta.yaml`:
+
+- `workspace.yaml` – workspace configuration. Do not forged to provide correct url to the `meta.yaml` file:
+
+  ```
+  metadata:
+    name: che-ideaiu
+  components:
+    - type: cheEditor
+      reference: '<url for the meta.yaml goes here>'
+      alias: theia-editor
+      automountWorkspaceSecrets: true
+  apiVersion: 1.0.0
+  ```
+
+  > Note, that in current workspace definition there is a new property: `automountWorkspaceSecrets: true`. This property instructs Eclipse Che to provision secrets into specific component. In our case into Che Editor based on IntelliJ Idea Ultimate Edition. This parameter is **mandatory** needed to successfully register IDE with activation code for offline usage.
+
+- `meta.yaml` – Che Editor configuration. Do not forget to replace `<username>` with the user name where image is pushed to:
+
+  ```
+  apiVersion: v2
+  publisher: <username>
+  name: intellij-ultimate-NOVNC
+  version: 2020.2.3
+  type: Che Editor
+  displayName:  IntelliJ IDEA Ultimate Edition
+  title:  IntelliJ IDEA Ultimate Edition (in browser using noVNC) as editor for Eclipse Che
+  description:  IntelliJ IDEA Ultimate Edition running on the Web with noVNC
+  icon: https://resources.jetbrains.com/storage/products/intellij-idea/img/meta/intellij-idea_logo_300x300.png
+  category: Editor
+  repository: https://github.com/che-incubator/che-editor-intellij-community
+  firstPublicationDate: "2020-10-27"
+  spec:
+    endpoints:
+    # hack (use dirigible endpoint name as it is part of liveness probes)
```

Comment:
now the hack is gone https://github.com/eclipse/che/blob/bedef8307f72c710c3344ae27910d39b64e07a9b/assembly/assembly-wsmaster-war/src/main/webapp/WEB-INF/classes/che/che.properties#L165

Reply:
Updated with `intellij`.